src/components/molecules/ExportProgressItem.tsx

```diff
@@ -0,0 +1,615 @@
+
+  const [dialogIsOpen, setDialogIsOpen] = useState(false)
+  const openDialog = useCallback(async () => {
+    if (dialogIsOpen) {
+      return
+    }
+    setDialogIsOpen(true)
+    try {
+      const result = await showOpenDialog({
+        properties: ['openDirectory', 'createDirectory'],
+        buttonLabel: t('folder.select'),
+        defaultPath: getPathByName('home'),
+      })
+      if (result.canceled) {
+        return
+      }
+      if (result.filePaths == null) {
+        return
+      }
+
+      return result.filePaths[0]
+    } catch (error) {
+      throw error
+    } finally {
+      setDialogIsOpen(false)
+    }
+  }, [dialogIsOpen, t])
+
+  const updateExportState = useCallback(
+    (exportOperationState: Partial<ExportOperationState>) => {
+      setExportState((state) => {
+        return {
+          ...state,
+          ...exportOperationState,
+        }
+      })
+    },
+    []
+  )
+
+  const resetExportingStates = useCallback(() => {
+    setShowExportProgress(false)
+    updateExportState({
+      exportOperation: '',
+      exportSubOperationTitle: '',
+      exportSubOperationContent: '',
+      exportProgressValue: 0,
+    })
+    setExportErrors([])
+  }, [updateExportState])
+
+  const addExportError = useCallback((error: string) => {
+    setExportErrors((state) => [...state, error])
+  }, [])
+
+  const cancelExporting = useCallback(() => {
+    canceledRef.current = true
+    resetExportingStates()
+  }, [resetExportingStates])
+
+  const isExportingCanceled = useCallback(() => {
+    return canceledRef.current
+  }, [])
+
+  const createExportDirectories = useCallback(
+    async (folderMap: ObjectMap<FolderDoc>, rootDir: string) => {
+      for (const folderPath of keys(folderMap)) {
+        if (isExportingCanceled()) {
+          return
+        }
+
+        if (folderPath == '/' && !exportSettings.exportingStorage) {
+          continue
+        }
+
+        if (
+          !exportSettings.exportingStorage &&
+          ((exportSettings.recursive &&
+            !folderPath.startsWith(exportSettings.folderPathname)) ||
+            (!exportSettings.recursive &&
+              folderPath != exportSettings.folderPathname))
+        ) {
+          continue
+        }
+
+        const fullPath = join(
+          rootDir,
+          folderPath.substring(folderPath.indexOf(exportSettings.folderName))
+        )
+
+        updateExportState({
+          exportSubOperationTitle: `Creating directory: ${fullPath.replace(
+            rootDir,
+            '.'
+          )}`,
+        })
+        await mkdir(fullPath)
+      }
+    },
+    [
+      exportSettings.exportingStorage,
+      exportSettings.folderName,
+      exportSettings.folderPathname,
+      exportSettings.recursive,
+      isExportingCanceled,
+      updateExportState,
+    ]
+  )
+
+  const convertValueToPercentage = useCallback(
+    (value: number, maxValue: number, exportProgressMaxValue = 100) => {
```

Comment:
I think we should use a single argument, `value/maxValue`, only rather than receiving the two arguments. If the value is NaN or Infinity, throw an error that it cannot be convertible to precentage

Reply:
Refactored!